### ISSUE TITLE
ubuntu failed to run

### ISSUE THREAD

**Issue by sxj111** [sxyu/volrend#3]
Is it due to insufficient RAM ?
![image](https://user-images.githubusercontent.com/27821863/114522239-c5e29680-9c75-11eb-80db-624f8cbf657e.png)

**Comment by sxyu** [comment]
Hello,
Sorry you're having trouble. We've been using Ubuntu the entire time, so I am fairly confident it should work. We're using OpenGL 4.3; perhaps your GPU does not support it? 

Could you post
- The result of: `glxinfo | grep version`
- Your Ubuntu version

**Comment by sxj111** [comment]
Thank you for your answer.
But I still have a question, how do I train to get files in npz format.
Will the training code be open source in the future?

**Comment by sxyu** [comment]
Hello,
We've released the training code in https://github.com/sxyu/plenoctree, feel free to try it!

**Comment by katealenic** [comment]
Hello, I had this error and change opengl version to 3.3 in code line, but now i get another error 
![image](https://user-images.githubusercontent.com/24976188/117135307-6e84a180-adaf-11eb-867f-8f035a83ef9a.png)
What can I do?

**Comment by sxyu** [comment]
hmm I got this errors like this when I was using my laptop, and it defaults to the integrated GPU that could be the issue. Please try
`glxinfo|egrep "OpenGL vendor|OpenGL renderer"`
and make sure it prints NVIDIA etc.

**Comment by katealenic** [comment]
Yes, l don't see nvidia, but I have it and nvidia-smi run correct
![image](https://user-images.githubusercontent.com/24976188/118259021-44bb3100-b4b9-11eb-8cab-1c6127ef3944.png)

**Comment by katealenic** [comment]
I use X-server to run and change in all place
glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
but I have this error
![image](https://user-images.githubusercontent.com/24976188/118284469-21a07980-b4d9-11eb-98e5-621f23ae7ff9.png)

**Comment by sxyu** [comment]
Honestly I am not sure I can help with this particular issue sorry
You could try to build the non-CUDA version right now, it's somewhat limited
`cmake .. -DVOLREND_USE_CUDA=OFF`
